src/main/webapp/app/exercises/file-upload/manage/file-upload-exercise.component.ts

```diff
@@ -48,12 +49,18 @@ export class FileUploadExerciseComponent extends ExerciseComponent {
                         exercise.course = this.course;
                         this.accountService.setAccessRightsForExercise(exercise);
                     });
+                    this.applyFilter();
                     this.emitExerciseCount(this.fileUploadExercises.length);
                 },
                 (res: HttpErrorResponse) => onError(this.alertService, res),
             );
     }
 
+    protected applyFilter(): void {
+        this.filteredFileUploadExercises = this.fileUploadExercises.filter((exercise) => this.filter.includeExercise(exercise));
+        this.emitFilteredExerciseCount(this.filteredFileUploadExercises.length);
+    }
+
```

Comment:
This method seems to appear in every component that is subclass of `ExerciseComponent`. Could we remove duplication and keep it in `ExerciseComponent` class?

Reply:
I don't really see a way to do this without bigger refactoring of all ExerciseComponent classes.
If you know a better way to handle this please let me know.